### PR TITLE
[Fix] RefreshControl Position Bug

### PR DESCRIPTION
**Pull request by GantMan**
Refresh bug #7976 
This bug has persisted for 3 versions of React Native.   Currently everyone is having to do a silly fix, documented in the bug.   I spent an hour trying to find the source of the bug.   Failing to find the issue quickly, I've just decided to make this simple bugfix.   

According to @janicduplessis #7976 we will likely be re-writing this control in iOS 10.

### CODE REVIEW

**Comment by facebook-github-bot-2** [comment]
By analyzing the blame information on this pull request, we identified @janicduplessis and @UnoDeTantos to be potential reviewers.

**Comment by GantMan** [comment]
Will update and fix.  I forgot you guys love the semicolon!

**Comment by facebook-github-bot-8** [comment]
@GantMan updated the pull request.

**Comment by facebook-github-bot-3** [comment]
@GantMan updated the pull request.

**Comment by facebook-github-bot-3** [comment]
@GantMan updated the pull request.

**Comment by facebook-github-bot-3** [comment]
@GantMan updated the pull request.

**Comment by janicduplessis** [comment]
Have you tried doing this fix in objc since it is only needed for iOS? Maybe try `if (backgroundColor == nil) backgroundColor = [UIColor clearColor];` somewhere in `RCTRefreshControl`.

If that doesn't work let's add a `Platform.OS` check in JS to make sure it only runs on iOS.

**Comment by GantMan** [comment]
sounds good, gonna check on that now.

**Comment by facebook-github-bot-4** [comment]
@GantMan updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9249/files/cd30e6f27dee0972804e54439c73ba09ebef3072..e1d9d292258e194f5249be442d815854b152e9e1)

**Comment by facebook-github-bot-4** [comment]
@GantMan updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9249/files/e1d9d292258e194f5249be442d815854b152e9e1..354e3a16632aa51a20a374b066ce2c488fa06187)

**Comment by GantMan** [comment]
@janicduplessis  - you were right.  The iOS fix works great.  This also helps keep the clutter out of JavaScript.  Tested this in a local app, it worked perfect.

**Comment by facebook-github-bot-7** [comment]
@GantMan updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9249/files/354e3a16632aa51a20a374b066ce2c488fa06187..b795887c9d7ccf0d0cace37a65bede530b5aa40f)

**Comment by facebook-github-bot-5** [comment]
@GantMan updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9249/files/b795887c9d7ccf0d0cace37a65bede530b5aa40f..ad9fa45485fd6ecaf7bedb0e4b120a9b23f3f8f1)

File: React/Views/RCTRefreshControl.m [hunk 1]
```diff
@@ -33,6 +33,9 @@ - (void)layoutSubviews
 {
   [super layoutSubviews];
 
+  // Quick fix for iOS backgroundColor bug #7976 - To be removed on update
+  if (self.backgroundColor == nil) self.backgroundColor = [UIColor clearColor];
```
Comment (janicduplessis) [inline React/Views/RCTRefreshControl.m]: Nit: put if content on new line with brackets.

File: React/Views/RCTRefreshControl.m [hunk 2]
```diff
@@ -33,6 +33,9 @@ - (void)layoutSubviews
 {
   [super layoutSubviews];
 
+  // Quick fix for iOS backgroundColor bug #7976 - To be removed on update
```
Comment (janicduplessis) [inline React/Views/RCTRefreshControl.m]: Can you change this comment to look like
// Fix for bug #...
// TODO: Remove when updating to use iOS 10 refreshControl ScrollView prop.

**Comment by janicduplessis** [comment]
Just a few comments inline then this is good to go :)

**Comment by facebook-github-bot-4** [comment]
@GantMan updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9249/files/ad9fa45485fd6ecaf7bedb0e4b120a9b23f3f8f1..4d9d80cdd59606c0ba97b316e98b404b5510bff0)

**Comment by GantMan** [comment]
Hey @janicduplessis feedback incorporated.  Sorry for the delay, I didn't see the comments in my notifications 👍 :shipit:

**Comment by facebook-github-bot-5** [comment]
@GantMan updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9249/files/4d9d80cdd59606c0ba97b316e98b404b5510bff0..191f47f607246899a9562cf2cbd5c97252d0b182)

**Comment by janicduplessis** [comment]
@GantMan Thanks!

@facebook-github-bot shipit

**Comment by facebook-github-bot-2** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://phabricator.intern.facebook.com/D3817227) to review internal test results.

**Pull request closed without merging**